internal/services/communication/communication_service_resource.go

```diff
@@ -141,6 +141,23 @@ func resourceArmCommunicationServiceRead(d *pluginsdk.ResourceData, meta interfa
 		d.Set("data_location", props.DataLocation)
 	}
 
+	keysResp, err := client.ListKeys(ctx, id.ResourceGroup, id.Name)
+
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[DEBUG] %s was not found - removing from state", *id)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+
+	d.Set("primary_connection_string", keysResp.PrimaryConnectionString)
+	d.Set("secondary_connection_string", keysResp.SecondaryConnectionString)
+	d.Set("primary_key", keysResp.PrimaryKey)
+	d.Set("secondary_key", keysResp.SecondaryKey)
```

Comment:
we also need to add these fields to the schema (above) as Computed strings - and then check that each field is set into the state in a test:

Test: https://github.com/hashicorp/terraform-provider-azurerm/blob/b9ac506edcc02c0ac049cd1f6e85205341c666c9/internal/services/communication/communication_service_resource_test.go#L26

via e.g.

> check.That(data.ResourceName).Key("primary_connection_string").Exists()